### PR TITLE
feat(voip-line): remove useless modal

### PR DESCRIPTION
**Pull request by Steffy29**
Ref: UXCT-23

### CODE REVIEW

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 1]
```diff
@@ -1,170 +1,205 @@
...
+        { value: 'external', label: this.$translate.instant('telephony_alias_configuration_agents_type_external') },
+      ];
+
+      this.onChooseServicePopover = ({ serviceName }, pos) => {
+        this.addAgentForm.numbers[pos] = serviceName;
+      };
+
+      this.$q.all({
+        translations: this.$translate.refresh(),
+        services: this.TelephonyMediator.getAll(),
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: You can use this service : https://github.com/ovh-ux/telecom-universe-components/blob/master/src/telecom/voip/telecom-voip.service.js#L52 to get all services.

For a better explanation, `TelephonyMediator` is meant to be replaced completely by cleaner service `voipService`.
(Also, just know that`voipService` and its children classes don't cover all existing features of TelephonyMediator))

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 2]
```diff
@@ -1,170 +1,205 @@
...
-        _.set(agent, 'inEdition', null);
-        this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_update_success'));
-      })
-      .catch((err) => {
-        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_update_error'), _.get(err, 'data.message')].join(' '));
+
+      return this.fetchAgentsIds().then((ids) => {
+        this.agents.ids = ids;
+      }).catch((err) => {
+        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_get_error'), _.get(err, 'data.message')].join(' '));
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: You can use template literal :
```suggestion
        this.TucToast.error(`${this.$translate.instant('telephony_alias_configuration_agents_get_error')} ${_.get(err, 'data.message', ''}`);
```

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 3]
```diff
@@ -1,170 +1,205 @@
...
-  addAgents() {
-    const modal = this.$uibModal.open({
-      animation: true,
-      templateUrl: 'app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx-modal.html',
-      controller: 'telecomTelephonyAliasConfigurationAgentsOvhPabxModal',
-      controllerAs: '$ctrl',
-    });
-    modal.result.then(() => {
+
+    orderBy(prop, asc) {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Maybe we can replace the table by a oui-datagrid, and get rid of this method no ? (https://github.com/ovh-ux/ovh-manager-telecom/blob/master/client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.html#L133)

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 4]
```diff
@@ -1,170 +1,205 @@
...
+            .remove({
+              billingAccount: this.$stateParams.billingAccount,
+              serviceName: this.$stateParams.serviceName,
+              agentId: id,
+            }).$promise.then(() => {
+              _.pull(this.agents.id, parseInt(id, 10));
+            })))
+        .then(() => {
+          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
+          return this.fetchAgentsIds().then((ids) => {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Better `return this.fetchAgentsIds()` and move the then block into a one at the same level than `catch` and `finally` for readability.

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 5]
```diff
@@ -1,170 +1,205 @@
...
+          serviceName: this.$stateParams.serviceName,
+          agentId: agent.agentId,
+        }, agent.inEdition).$promise
+        .then(() => {
+          _.assign(agent, agent.inEdition);
+          _.set(agent, 'inEdition', null);
+          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_update_success'));
+        })
+        .catch((err) => {
+          this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_update_error'), _.get(err, 'data.message')].join(' '));
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Same thing (template literal)

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 6]
```diff
@@ -1,170 +1,205 @@
...
+      } else if (this.addAgentForm.numbers.length > 1) {
+        _.pullAt(this.addAgentForm.numbers, pos);
+      } else {
+        this.addAgentForm.numbers[0] = null;
+      }
+    }
+
+    // Check if external numbers are defined into the list
+    checkExternalNumber() {
+      const filter = this.addAgentForm.numbers.filter(number => this.groupList
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]:
```suggestion
      const filteredNumbers = this.addAgentForm.numbers.filter(number => this.groupList
```

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 7]
```diff
@@ -1,170 +1,205 @@
...
+          .filter(service => service.serviceType === 'alias')
+          .filter(service => service.serviceName === number)
+          .length > 0)
+        .length > 0);
+      return filter.length !== this.addAgentForm.numbers.length;
+    }
+
+    addAgents() {
+      if (this.checkExternalNumber()) {
+        const modal = this.$uibModal.open({
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: You can get rid of `modal` var : 
```suggestion
        this.$uibModal.open(myModal).result.then(() => etc....)
```

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 8]
```diff
@@ -181,21 +216,17 @@ angular.module('managerApp').controller('TelecomTelephonyAliasConfigurationAgent
...
-        .catch((err) => {
-          this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_add_error'), _.get(err, 'data.message')].join(' '));
-          return this.$q.reject(err);
-        })
-        .finally(() => {
-          this.addAgentForm.isAdding = false;
-          this.addAgentForm.numbers = [null];
+      })).then(() => {
+        this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_add_success'));
+        return this.fetchAgentsIds().then((ids) => {
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Same thing for returning promise and move `then` block below, at the same level of `$q.all`

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js [hunk 9]
```diff
@@ -181,21 +216,17 @@ angular.module('managerApp').controller('TelecomTelephonyAliasConfigurationAgent
...
+          this.agents.ids = ids;
         });
-    });
-  }
-});
+      }).catch((err) => {
+        this.TucToast.error([this.$translate.instant('telephony_alias_configuration_agents_add_error'), _.get(err, 'data.message')].join(' '));
```
Comment (JeremyDec) [inline client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Template literal

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js (hunk 4 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Sorry but I don't understand your feedback on this return. Could you explain it to me?

Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: Sure I mean to do this, to improve readability

```suggestion
.then(() => {
          this.TucToast.success(this.$translate.instant('telephony_alias_configuration_agents_delete_success'));
          return this.fetchAgentsIds();
})
.then((ids) => { 
   // Your code
})
```

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js (hunk 1 above)
Comment (JeremyDec) [reply client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: `this.tucTelecomVoip.fetchAll()` returns you all billingAccounts with services in.
So `TelephonyMediator.groups` can be replaced by `result.services` (Should be rename as `result.billingAccounts` for clarity)

File: client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js (hunk 3 above)
Comment (Steffy29) [reply client/app/telecom/telephony/alias/configuration/agents/ovhPabx/telecom-telephony-alias-configuration-agents-ovhPabx.controller.js]: I note this feedback for a future issue. For the moment, it needs to refactor a big part of the table to replace with oui-datagrid.

**Pull request merged**